### ISSUE TITLE
Make code snippets from docstrings run by  doctest 

### ISSUE THREAD

**Issue by bsipocz** [astropy/astropy#9485]
https://github.com/astropy/astropy/pull/9468 highlighted that there are code snippets in docstrings that are not actually picked up during doctesting. I strongly feel that shouldn't be the case, we should skip any such testing in very rare occasions.
I tag this as a coordinates issue, but they may be other sub packages affected by the same problem.

**Comment by pllim** [comment]
What is the solution you have in mind? Prepend `>>>` in front of them to be picked up by `doctest`?

**Comment by bsipocz** [comment]
Basically yes. 
But then they most probably needs to be extended to have all the required imports. So overall change them from many one liners to a few few liners.

**Comment by Harshil-C** [comment]
hey ,iam a student and i want  to work on this issue as a step to familiarise the project

**Comment by mhvk** [comment]
@Harshil-C - that's great! (Sorry for the slow response, we're in the middle of a new release.) Please just go ahead and try to fix some. It will be by far easiest if you set up your own computer such that it can do the testing (see help for developpers in the main documentation page). But if you are unsure, do feel free to ask - perhaps easiest based on a PR that changes at least a few examples to doctests.

**Comment by Harshil-C** [comment]
thanks for replying ,
should i run the doctest and find the missing code snippets just on the  sky_cooordinate.py   file

**Comment by mhvk** [comment]
Sure, start with `sky_coordinate.py`!

**Comment by DShivansh** [comment]
@mhvk can I work on this issue, if no one is working on it ?

**Comment by mhvk** [comment]
@DShivansh - it doesn't seem @Harshil-C actually did it, so, yes, go ahead (@Harshil-C, please give a shout if you have something ready).

**Comment by DShivansh** [comment]
@mhvk can you please show me the code or tell me the file in which I can find it as it seems I am not able to fully understand the problem correctly.

**Comment by mhvk** [comment]
@DShivansh - the issue here is that #9468 found that some examples were not marked as doctests and thus not ran in tests, and that we want to be sure there are no others. A quick look at `sky_coordinate.py` actually suggests there no other problem cases (though I didn't look in detail), but in a way the work here is to look through the code and make sure that is true for the whole module. If it turns out we have been otherwise good and everything is a doctest (i.e., has `>>>` in front of every piece of example code), then that is very useful to know too!

**Comment by Harshil-C** [comment]
i coudnt find  any  statement in  the example to be prepended ,but idid find that while running  doctest to the module it showed up a value error that teminated the tests .I tried the test removing the statement to check whether  it because of that partiular statement and the doctest did pass all the test.
As a novice i didnt understand the purpose of the comment statement # doctest: +FLOAT_CMP  
this value error showed up when the sign ':' came up,i think it kills the comment which gives the value error.
i think that is th reason ,and it rally help if i could know the meaning of # doctest: +FLOAT_CMP

**Comment by Harshil-C** [comment]
the part of the code
        >>> c1 = SkyCoord(0*u.deg, 0*u.deg)
        >>> c2 = SkyCoord(1*u.deg, 0*u.deg)
        >>> c1.position_angle(c2).degree
        90.0
        >>> c3 = SkyCoord(1*u.deg, 1*u.deg)
        >>> c1.position_angle(c3).degree         # doctest: +FLOAT_CMP
        44.995636455344844

**Comment by Harshil-C** [comment]
THE VALUE ERROR SHOWED
Traceback (most recent call last):
  File "C:\Users\asus\Desktop\astropy\astropy\coordinates\sky_coordinate.py", line 1707, in <module>
    doctest.testmod()
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 1947, in testmod
    for test in finder.find(m, name, globs=globs, extraglobs=extraglobs):
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 932, in find
    self._find(tests, obj, name, module, source_lines, globs, {})
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 994, in _find
    self._find(tests, val, valname, module, source_lines,
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 1028, in _find
    self._find(tests, val, valname, module, source_lines,
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 982, in _find
    test = self._get_test(obj, name, module, globs, source_lines)
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 1065, in _get_test
    return self._parser.get_doctest(docstring, globs, name,
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 668, in get_doctest
    return DocTest(self.get_examples(string, name), globs,
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 682, in get_examples
    return [x for x in self.parse(string, name)
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 644, in parse
    self._parse_example(m, name, lineno)
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 725, in _parse_example
    options = self._find_options(source, name, lineno)
  File "C:\Users\asus\AppData\Local\Programs\Python\Python38-32\lib\doctest.py", line 754, in _find_options
    raise ValueError('line %r of the doctest for %s '
ValueError: line 27 of the doctest for __main__.SkyCoord.position_angle has an invalid option: '+FLOAT_CMP'

**Comment by Harshil-C** [comment]
THE SUCCESSFUL DOCTEST AFTER THE REMOVAL OF THE  # doctest: +FLOAT_CMP
Trying:
    c1.position_angle(c2).degree
Expecting:
    90.0
ok
Trying:
    c3 = SkyCoord(1*u.deg, 1*u.deg)
Expecting nothing
ok
Trying:
    c1.position_angle(c3).degree
Expecting:
    44.995636455344844
ok

2 items passed all tests:
  20 tests in sky_coordinate.SkyCoord
   5 tests in sky_coordinate.SkyCoord.position_angle
25 tests in 40 items.
25 passed and 0 failed.
Test passed.

**Comment by Harshil-C** [comment]
FLOAT_CMP flag appeared in pytest-doctestplus 0.5,  https://github.com/astropy/pytest-doctestplus/commit/0cebe90

**Comment by Harshil-C** [comment]
i think if it is a comment the ':' symbol causes a problem here

**Comment by ayshih** [comment]
I suspect that you either don't have `pytest-doctestplus` installed or have an old version.  `FLOAT_CMP` is an optional directive provided by `pytest-doctestplus` that enables nice comparison of floats.  The syntax of `# doctest: +FLOAT_CMP` is correct (including the colon), and should work on a setup with a proper version of `pytest-doctestplus`.

**Comment by Harshil-C** [comment]
oh so chances  are that i dont have pytest doctest  plus installed
 thanks man i didnt know about that doctest module

and sorry for all the misleading blunder suggestion

**Comment by Harshil-C** [comment]
oh so my chances are that i didnt have the  pytest doctest plus installed 
thanks man i didnt knew about that module until now .

and sorry for all the misleading blunder suggestions

**Comment by Harshil-C** [comment]
@mhvk  after many look through i couldnt find any example lacking the >>> prepend .i would like to know how @bsipocz  tells that there is missing examples ,i mean is there a astropy test suite argument that i must know or @bsipocz  just knows that from memory.(let me know if i said anything wrong)

**Comment by Harshil-C** [comment]
and i was checking the whole coordinates package by looking the docs rst file and found that angle_space_motion  dos have >>>prepend missing code snippets but iam not sure whether to consider it as a docstring .here is screen shot of the documentation .
![Screenshot (1)](https://user-images.githubusercontent.com/57037666/72446423-8ced1500-37d9-11ea-8efc-09fa043b7da2.png)

**Comment by Harshil-C** [comment]
@mhvk  @bsipocz i ran the test with astropy test suit(mostly from the python interpreter) .
i started with sky_coordinate.py and i couldnt find any >>> prepend needed ,then i moved onto the whole coordinate package and found some suscpicious docstring code snippet that look like it is missing out the doctests .i wanna speacify thef iles and the respective lines where the code snippet is .If you are busy i could just put the image here  on request .Here are the files:-
*solar_system.py  line.no:74  
*transformations.p line.no:651
*attributes.py line.no:28   
*baseframe.py line.no:662
*errors.py lin.nos: 36,82,123

i must confirm that is what this issue is about from you people,so i can do the nessaccry change in code.
(tell me if this is wrong)
iam not confident about it cause iam confused that this issue deals with docskips  or not

**Comment by nikita-0209** [comment]
@mhvk Hi, I am a beginner.  If no one else is, I want to work on this issue. 
I went through [sky_coordinate.py](https://github.com/astropy/astropy/blob/master/astropy/coordinates/sky_coordinate.py)
I found code snippets in docstrings only in `Position_Angle`. Is that all we are supposed to test?

**Comment by mhvk** [comment]
@nikita-0209 - I'm actually not sure there still are issues that need looking at. @bsipocz - do you know? Or should we close this issue?

**Comment by bsipocz** [comment]
I think most of these got indeed cleaned up, but there may some more instances around. E.g one could argue this list ideally should be sent through doctest with ignoring output. A few more files with more examples (most are borderline) are listed below based on a quick and dirty grep:

```
class Angle(u.SpecificTypeQuantity):
    """
    One or more angular value(s) with units equivalent to radians or degrees.

    An angle can be specified either as an array, scalar, tuple (see
    below), string, `~astropy.units.Quantity` or another
    :class:`~astropy.coordinates.Angle`.

    The input parser is flexible and supports a variety of formats::

      Angle('10.2345d')
      Angle(['10.2345d', '-20d'])
      Angle('1:2:30.43 degrees')
      Angle('1 2 0 hours')
      Angle(np.arange(1, 8), unit=u.deg)
      Angle('1°2′3″')
      Angle('1°2′3″N')
      Angle('1d2m3.4s')
      Angle('1d2m3.4sS')
      Angle('-1h2m3s')
      Angle('-1h2m3sE')
      Angle('-1h2.5m')
      Angle('-1h2.5mW')
      Angle('-1:2.5', unit=u.deg)
      Angle((10, 11, 12), unit='hourangle')  # (h, m, s)
      Angle((-1, 2, 3), unit=u.deg)  # (d, m, s)
      Angle(10.2345 * u.deg)
      Angle(Angle(10.2345 * u.deg))
```
see also `astropy/io/ascii/latex.py`, `astropy/time/core.py`, `astropy/table/column.py `, `astropy/io/ascii/misc.py`

**Comment by nikita-0209** [comment]
I've created a Pull Request #10261

**Comment by astrojuanlu** [comment]
Prospective contributors: see [end of @bsipocz comment above](https://github.com/astropy/astropy/issues/9485#issuecomment-622005801) for a list of modules that need to be checked.

**Comment by zupeiza** [comment]
I'll work in the modules above:

- astropy/io/ascii/latex.py
- astropy/time/core.py
- astropy/table/column.py
- astropy/io/ascii/misc.py

**Comment by astrojuanlu** [comment]
With #10261 (merged) and #10866 (pending review) I _think_ we can close this issue.

**Comment by pllim** [comment]
Sure. We can address any future cases that appear in a "as needed" basis. Thanks for following up on this!
